src/controllers/ElementsController.php

```diff
@@ -884,25 +884,52 @@ private function _additionalButtons(
 
         // Apply draft
         if ($isDraft && !$isCurrent && $canSave && $canSaveCanonical) {
-            $components[] = Html::button(Craft::t('app', 'Apply draft'), [
-                'class' => ['btn', 'secondary', 'formsubmit'],
+            /** @phpstan-ignore-next-line */
+            $disabled = $canonical->hasMethod('isEntryTypeCompatible') && !$element->isEntryTypeCompatible();
```

Comment:
ElementsController shouldn’t be giving any special treatment to entries, so we’ll need to rethink how we do this.